mixins/createGroupRouteMixin.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A function to create a mixin to be used where there is an
+ * optional group param in the route.
+ *
+ * Designed to be used with the GroupSelect component.
+ *
+ * @param key used for the remember key and the route segment
+ * @returns a Vue mixin
+ */
+export default function createGroupRouteMixin(key) {
+  const rememberId = key
+  function routePath(id) {
+    return `/${key}/` + (id === 0 ? '' : id)
+  }
+  return {
+    computed: {
+      groupid: {
+        get() {
+          return this.$route.params.id ? parseInt(this.$route.params.id) : 0
+        },
+        set(val) {
+          const oldVal = this.groupid
+          val = val || 0
+          if (val !== oldVal) {
+            // We have changed the groupid away from the one in the route! Redirect...
+            this.$router.push(routePath(val))
+            if (val === 0) {
+              this.$store.commit('group/forget', {
+                id: rememberId
+              })
+            }
+          }
+        }
+      }
+    },
+    beforeDestroy() {
+      if (this._unwatchGroupRemember) this._unwatchGroupRemember()
+    },
+    created() {
+      this._unwatchGroupRemember = this.$store.watch(
+        (state, getters) => getters['group/remembered'](rememberId),
+        (val, oldVal) => {
+          if (oldVal === undefined && val === undefined && this.groupid !== 0) {
+            // Nothing set so far... make it what our current page is
+            this.$store.commit('group/remember', {
+              id: rememberId,
+              val: this.groupid
+            })
+          } else if (val !== undefined) {
+            if (this.groupid === 0) {
+              // We have a remember value, but we're on the general page
+              // Replace the current route
+              this.$router.replace(routePath(val))
```

Comment:
This is the magic method to change the route without adding it into the back stack....